unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,217 @@
+
+    def test_check_buffer_message_matching_command_and_id_throws_another_exception(self, redis_client_auto_register):
+        """checks buffer message matching command and id throws exception"""
+        message = {"data": {"transaction_id": TRANSACTION_ID}}
+        with pytest.raises(RedisAPIException,
+                           match="There is no matching command response"
+                                 " in _blocking_command_responses."):
+            redis_client_auto_register \
+                ._check_buffer_message_matching_command_and_id(message["data"])
+
+    def test_register(self, redis_client_auto_register):
+        """checks whether the redis.db.publish is called with correct arguments"""
+        data = {"name": AREA_ID, "transaction_id": AREA_ID}
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.register(is_blocking=False)
+        redis_client_auto_register.redis_db.publish \
+            .assert_called_once_with(f"{AREA_ID}/register_participant", json.dumps(data))
+
+    def test_register_blocking_true_throws_exception(self, redis_client_auto_register):
+        """checks whether the if is blocking is set to true throws exception"""
+        with pytest.raises(RedisAPIException,
+                           match="API registration process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background "
+                                 "and will notify you as soon as the "
+                                 "registration has been completed."):
+            redis_client_auto_register.register(is_blocking=True)
+
+    def test_register_blocking_false_throw_exception(self, redis_client_auto_register):
+        """checks whether the if is blocking is set to false throws exception"""
+        with pytest.raises(RedisAPIException,
+                           match="API is already registered to the market."):
+            self.redis_client_base = redis_client_auto_register
+            self.redis_client_base.is_active = True
+            self.redis_client_base.register(is_blocking=False)
+
+    def test_unregister(self, redis_client_auto_register):
+        """checks whether the redis.db.publish is called with correct arguments"""
+        data = {"name": AREA_ID, "transaction_id": AREA_ID}
+        self.redis_client = redis_client_auto_register
+        self.redis_client.redis_db.publish = MagicMock()
+        self.redis_client.is_active = True
+        self.redis_client.unregister(is_blocking=False)
+        self.redis_client.redis_db.publish \
+            .assert_called_once_with(f"{AREA_ID}/unregister_participant",
+                                     json.dumps(data))
+
+    def test_unregister_blocking_is_active_false_throws_exception(self, redis_client_auto_register):
+        """checks if is active set to false throws exception"""
```

Comment:
https://github.com/gridsingularity/d3a-api-client/pull/163/files#r695721335 applies here too (only that the user is not already registered)